### PR TITLE
fix: execution names for testsuites

### PR DESCRIPTION
**Pull request by nicufk**
## Checklist (choose whats happened)

- [ ] breaking change! (describe)
- [ ] tested locally
- [ ] tested on cluster
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

## Breaking changes

-

## Changes

-

## Fixes

-

### CODE REVIEW

File: api/v1/testkube.yaml [hunk 1]
```diff
@@ -2675,6 +2675,9 @@ components:
           type: string
           description: test execution custom name
           example: "testing with 1000 users"
+        number:
+          type: integer
+          description: test suite execution number
```
Comment (vLia) [inline api/v1/testkube.yaml:2680]: would be nice to have an example here for the docs

**Review by vLia — APPROVED** [review]
looks good! had one tiny comment

File: internal/app/api/v1/testsuites.go [hunk 2]
```diff
@@ -587,9 +587,11 @@ func (s TestkubeAPI) GetTestSuiteExecutionHandler() fiber.Handler {
...
 		id := c.Params("executionID")
 		execution, err := s.TestExecutionResults.Get(ctx, id)
-
-		if err != nil {
-			return s.Error(c, http.StatusBadRequest, err)
+		if err == mongo.ErrNoDocuments {
+			execution, err = s.TestExecutionResults.GetByName(ctx, id)
+			if err == mongo.ErrNoDocuments {
+				return s.Error(c, http.StatusNotFound, fmt.Errorf("test with execution id/name %s not found", id))
+			}
```
Comment (vsukhin) [inline internal/app/api/v1/testsuites.go:594]: If it's a different error type?

File: internal/app/api/v1/testsuites.go [hunk 3]
```diff
@@ -649,12 +651,11 @@ func (s TestkubeAPI) executeTestSuite(ctx context.Context, testSuite testkube.Te
 		return testsuiteExecution, err
 	}
 
+	request.Number = s.getNextExecutionNumber("ts-" + testSuite.Name)
+	request.Name = fmt.Sprintf("ts-%s-%d", testSuite.Name, request.Number)
+
 	request.SecretUUID = secretUUID
 	if testSuite.ExecutionRequest != nil {
-		if request.Name == "" && testSuite.ExecutionRequest.Name != "" {
```
Comment (vsukhin) [inline internal/app/api/v1/testsuites.go:654]: Should we ignore, if user sends this name in test suite crd?

**Review by vsukhin — COMMENTED** [review]
a couple of questions

**Review by vsukhin — APPROVED** [review]

**Pull request merged**